src/neuroconv/tools/signal_processing.py

```diff
@@ -24,7 +24,7 @@ def get_rising_frames_from_ttl(trace: np.ndarray, threshold: Optional[float] = N
     threshold = threshold if threshold is not None else np.mean(trace)
 
     sign = np.sign(trace - threshold)
-    diff = np.diff(sign)
+    diff = np.diff(sign, axis=0)
```

Comment:
Needed to add this due to odd issue with 1-D input returning an empty diff

Reply:
Or I guess it was a 2-D input with the second dimension being '1' instead of open.

Alternatively, could `squeeze` the trace on passing?

Reply:
either way is fine. This will always be 1-d, right? You could also use .ravel()

Reply:
> This will always be 1-d, right?

As we use it internally, yes.

Technically someone could call the function directly and pass a `trace` with multiple dimensions; since we do not assert, check, or otherwise force it here.

Should I add such a check to enforce our intended usage of the function?

Reply:
`np.ravel` makes usage a bit clearer at no cost to performance